From 6b53bf765e953a21af518e05ab94334bde9bbb87 Mon Sep 17 00:00:00 2001
From: Jiyong Park <jiyong@google.com>
Date: Fri, 19 Aug 2022 13:07:48 +0900
Subject: [PATCH] Use liblog_for_runtime_apex instead of liblog

liblog_for_runtime_apex is a static variant of liblog which is
explicitly marked as available to the runtime APEX. Any static
dependency to liblog from inside the runtime APEX is changed from liblog
to liblog_for_runtime_apex.

Previously, to support the need for using liblog inside the runtime
APEX, the entire (i.e. both static and shared variants) liblog module
was marked as available to the runtime APEX, although in reality only
the static variant of the library was needed there. This was not only
looking dirty, but also has caused a problem like b/241259844.

To fix this, liblog is separated into two parts. (1) liblog and (2)
liblog_for_runtime_apex. (1) no longer is available to the runtime APEX
and is intended to be depended on in most cases: either from the
non-updatable platform, or from other APEXes. (2) is a static library
which is explicitly marked as available to the runtime APEX and also
visible to certain modules that are included in the runtime APEX.

Bug: 241259844
Test: m and check that liblog depends on stub library of libc
Change-Id: I7a95ce26147fca27cc7b6a4662d3617b3f0092ec
---
 Android.bp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Android.bp b/Android.bp
index 8c3ae6d..f386b97 100644
--- a/Android.bp
+++ b/Android.bp
@@ -29,7 +29,7 @@ cc_defaults {
     static_libs: [
         "libapexutil",
         "libbase",
-        "liblog",
+        "liblog_for_runtime_apex",
         "libprotobuf-cpp-lite",
         "libxml2",
         "lib_linker_config_proto_lite",
-- 
2.38.1

